### DISCUSSION TITLE
The "for" in SAWScript

### DISCUSSION THREAD

**Discussion by LVPGroup** [GaloisInc/saw-script#1683] [Q&A]
"for [1..100] dowork" does not work in SAWScript, however "for [1,2,3,4] dowork" works. How can I write a loop from 1 to 100? 
Thanks.

**Comment by chameco** [comment]
It's a bit awkward, but you can use the `eval_list` function for this purpose alongside a Cryptol range.

For example:
```
for (eval_list {{ [1..100 : [8]] }}) (\x -> dowork (eval_int x));
```

**Reply by LVPGroup** [reply]
Thanks a lot. 
Actually, I am specifying a matrix with each element a pointer to a linked list, "data *matrix[32][32]", where each linked list is unbounded. As I know, in SAWScript we have to use a bounded array to represent a linked list, right? 

For the matrix, I use 

matrix <- llvm_fresh_var "matrix" (llvm_array 32 (llvm_array 32 (llvm_type "%struct.data*")));
free_blks <- llvm_fresh_var "free_blks" (llvm_array 32 (llvm_array 32 (llvm_array 1024 (llvm_alias "struct.data"))));  // assume bounded

and then I assign the pointer to each1024 array to each element of the matrix. Am I right?
